gramps/gen/lib/json_utils.py

```diff
@@ -0,0 +1,219 @@
+#
+# Gramps - a GTK+/GNOME based genealogy program
+#
+# Copyright (C) 2017,2024       Nick Hall
+# Copyright (C) 2024-2025       Doug Blank
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+#
+
+# ------------------------------------------------------------------------
+#
+# Python modules
+#
+# ------------------------------------------------------------------------
+
+import json
+import orjson
+
+# ------------------------------------------------------------------------
+#
+# Gramps modules
+#
+# ------------------------------------------------------------------------
+
+import gramps.gen.lib as lib
+
+
+NoneType = type(None)
+
+
+class DataDict(dict):
+    """
+    A wrapper around a data dict that also provides an
+    object interface.
+    """
+
+    def __init__(self, data=None):
+        """
+        Wrap a data dict (raw data) or object
+        with an attribute API. If data is an
+        object, we use it to get the attributes.
+        """
+        if isinstance(data, dict):
+            super().__init__(data)
+        else:
+            super().__init__()
+            if data:
+                # Data is actually a Gramps object
+                self.update(object_to_data(data))
+
+    def __str__(self):
+        if "_object" not in self:
+            self["_object"] = data_to_object(self)
```

Comment:
I'm not sure it is a good idea to cache the object here...  I was fooling around in my debugger and changed an element of the DataDict; only to be very confused when the debugger refused to show that the value had changed.  After looking at this I realized that the debugger was always showing the original values from the DataDict, not the current values.

Either this should not be cached, or we need to document that values in this dict should not be changed.

Reply:
That's a good point about having a cached version that is out of scope with a changed object. But all of the DataDict would be old too.